forms/htmleditor/TinyMCEConfig.php

```diff
@@ -416,31 +435,26 @@ protected function getEditorCSS() {
 	 * Generate gzipped TinyMCE configuration including plugins and languages.
 	 * This ends up "pre-loading" TinyMCE bundled with the required plugins
 	 * so that multiple HTTP requests on the client don't need to be made.
+	 *
+	 * @return string
 	 */
-	public function requireJS() {
-		require_once THIRDPARTY_PATH . '/tinymce/tiny_mce_gzip.php';
-
+	public function getScriptURL() {
+		// If gzip is disabled just return core script url
 		$useGzip = Config::inst()->get('HtmlEditorField', 'use_gzip');
-		$languages = array();
-
-		foreach(self::$configs as $configID => $config) {
-			$languages[] = $config->getOption('language');
+		if(!$useGzip) {
+			return THIRDPARTY_DIR . '/tinymce/tinymce.min.js';
 		}
 
 		// tinyMCE JS requirement
-		if($useGzip) {
-			$tag = TinyMCE_Compressor::renderTag(array(
-				'url' => THIRDPARTY_DIR . '/tinymce/tiny_mce_gzip.php',
-				'plugins' => implode(',', array_keys($this->getPlugins())),
-				'themes' => $this->getTheme(),
-				'languages' => implode(',', array_filter($languages))
-			), true);
-			preg_match('/src="([^"]*)"/', $tag, $matches);
-
-			Requirements::javascript(html_entity_decode($matches[1]));
-		} else {
-			Requirements::javascript(THIRDPARTY_DIR . '/tinymce/tinymce.min.js');
-		}
+		require_once THIRDPARTY_PATH . '/tinymce/tiny_mce_gzip.php';
+		$tag = TinyMCE_Compressor::renderTag(array(
+			'url' => THIRDPARTY_DIR . '/tinymce/tiny_mce_gzip.php',
+			'plugins' => implode(',', $this->getInternalPlugins()),
```

Comment:
Only minify plugins specified without a custom path, since you can't pass paths to the minifier.